src/app/public/testing/dropdown/dropdown-fixture.ts

```diff
@@ -0,0 +1,187 @@
+      horizontalAlignment: buttonCmp.horizontalAlignment,
+      label: buttonCmp.label,
+      title: buttonCmp.title
+    };
+  }}
+
+  /**
+   * Returns the dropdown button's text.
+   */
+  public get dropdownButtonText(): string {
+    return this.buttonDebugElement.nativeElement.innerText.trim();
+  }
+
+  /**
+   * Retrieves information about the dropdown menu component.
+   */
+  public get dropdownMenu(): SkyDropdownFixtureDropdownMenu {
+    const menu = this.getOverlay().querySelector('.sky-dropdown-menu');
+    if (!menu) {
+      return undefined;
+    }
+
+    return {
+      ariaLabelledBy: menu.getAttribute('aria-labelledby'),
+      ariaRole: menu.getAttribute('role')
+    };
+  }
+
+  /**
+   * Indicates if the dropdown menu is open and visible.
+   */
+  public get dropdownMenuIsVisible(): boolean {
+    const overlay = this.getOverlay();
+
+    if (!overlay) {
+      /* tslint:disable-next-line:no-null-keyword */
+      return false;
+    }
+
+    return this.getOverlay().querySelector('.sky-dropdown-menu') !== null;
+  }
+
+  private get buttonDebugElement(): DebugElement {
+    return this.debugEl.query(
+      By.css('.sky-dropdown-button')
+    );
+  }
+
+  constructor(
+    private fixture: ComponentFixture<any>,
+    skyTestId: string
+  ) {
+    this.debugEl = SkyAppTestUtility.getDebugElementByTestId(fixture, skyTestId, 'sky-dropdown');
+  }
+
+  /**
+   * Click the dropdown button to open or close the dropdown menu.
+   */
+  public async clickDropdownButton(): Promise<any> {
+    this.buttonDebugElement.nativeElement.click();
+    this.fixture.detectChanges();
+    return this.fixture.whenStable();
+  }
+
+  /**
+   * Click the dropdown item at the provided index.
+   */
+  public async clickDropdownItem(index: number): Promise<any> {
+    const itemEls = this.getDropdownItemEls();
+    if (index >= itemEls.length) {
+      throw new Error(`There is no dropdown item at index ${index}.`);
+    }
+
+    itemEls[index].click();
+
+    this.fixture.detectChanges();
+    return this.fixture.whenStable();
+  }
+
+  /**
+   * Retrieves information about the dropdown item at the provided index.
+   */
+  public getDropdownItem(index: number): SkyDropdownFixtureDropdownItem {{
+    const item = this.getDropdownItemEls()[index];
+    if (!item) {
+      return undefined;
+    }
+
+    return {
+      ariaRole: item.getAttribute('role')
+    };
+  }}
+
+  /**
+   * Returns the first element inside the dropdown menu
+   * that is a descendant of the provided selector.
+   */
+  public querySelector(selector: string): any {
+    const overlay = this.getOverlay();
+    if (!overlay) {
+      /* tslint:disable-next-line:no-null-keyword */
+      return null;
+    }
+
+    return overlay.querySelector(selector);
+  }
+
+  /**
+   * Returns all elements inside the dropdown menu
+   * that are descendants of the provided selector.
+   */
+  public querySelectorAll(selector: string): NodeListOf<any> {
```

Comment:
Didn't we want to throw an error if the selector included `sky-` or `.sky-`?

Reply:
Thanks for the reminder. I've added this.

Reply:
This logic might need to go into `SkyAppTestUtility` so it can be shared across other component fixtures.

Reply:
Done: https://github.com/blackbaud/skyux-sdk-testing/pull/33

Reply:
Circling back on this... After some further discussion, we decided adding `querySelector` to the public API would add confusion of when to use it vs the Web API's `querySelector`. This logic will still be handled in the fixture, but slightly differently. I've added a `getDropdownMenuContent()` method to the fixture that allows consumers to get at their template.